astropy/wcs/tests/test_tab.py

```diff
@@ -0,0 +1,37 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import pytest
+
+import numpy as np
+
+from astropy import wcs
+
+from . helper import SimModelTAB
+
+
+def test_2d_spacial_tab_roundtrip(tab_wcs_2di):
+    nx, ny = tab_wcs_2di.pixel_shape
+    # generate random test coordinates:
+    xy = 0.51 + [nx + 0.99, ny + 0.99] * np.random.random((100, 2))
+    rd = tab_wcs_2di.wcs_pix2world(xy, 1)
+    xy_roundtripped = tab_wcs_2di.wcs_world2pix(rd, 1)
+    m = np.logical_and(*(np.isfinite(xy_roundtripped).T))
+    assert np.allclose(xy[m], xy_roundtripped[m], rtol=0, atol=1e-7)
+
+
+def test_2d_spacial_tab_vs_model():
+    nx = 150
+    ny = 200
+    model = SimModelTAB(nx=nx, ny=ny)
+
+    # generate FITS HDU list:
+    hdulist = model.hdulist
+
+    # create WCS object:
+    w = wcs.WCS(hdulist[0].header, hdulist)
+
+    # generate random test coordinates:
+    xy = 0.51 + [nx + 0.99, ny + 0.99] * np.random.random((100, 2))
```

Comment:
@mcara says there is no need for random
```suggestion
    xy = 0.51 + [nx + 0.99, ny + 0.99] * np.zeros((100, 2))
```